x-pack/plugins/uptime/public/components/queries/filter_bar/filter_bar.tsx

```diff
@@ -74,13 +69,35 @@ export const FilterBar = ({ dateRangeEnd, dateRangeStart, updateQuery }: Props)
           }),
           multiSelect: false,
           options: ids
-            ? take(ids, MAX_SELECTION_LENGTH).map(({ key, url }: any) => ({
+            ? ids.map(({ key }: MonitorKey) => ({
                 value: key,
-                view: url,
+                view: key,
               }))
             : [],
           searchThreshold: SEARCH_THRESHOLD,
         },
+        {
+          type: 'field_value_selection',
+          field: 'monitor.name',
+          name: i18n.translate('xpack.uptime.filterBar.options.nameLabel', {
+            defaultMessage: 'Name',
+          }),
+          multiSelect: false,
```

Comment:
Yeah - the thought has crossed my mind. I don't know how much more work it'd be because we're just specifying a JSON string. Perhaps a more secure way to handle this, however, would be to define API types in our GQL schema and expect those. We'd be able to get rid of some ugly server code, and it would be more secure than accepting arbitrary JSON strings.

Reply:
++